resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -535,18 +553,29 @@ private InetSocketAddress nextNameServerAddress() {
             InetSocketAddress nameServerAddr, DnsQuestion question,
             Promise<AddressedEnvelope<? extends DnsResponse, InetSocketAddress>> promise) {
 
-        return query0(checkNotNull(nameServerAddr, "nameServerAddr"),
-                      checkNotNull(question, "question"),
-                      checkNotNull(promise, "promise"));
+        return query0(nameServerAddr, question, Collections.<DnsRecord>emptyList(), promise);
+    }
+
+    /**
+     * Sends a DNS query with the specified question with additional records using the specified name server list.
+     */
+    public Future<AddressedEnvelope<DnsResponse, InetSocketAddress>> query(
+            InetSocketAddress nameServerAddr, DnsQuestion question,
+            List<DnsRecord> additional,
+            Promise<AddressedEnvelope<? extends DnsResponse, InetSocketAddress>> promise) {
+
+        return query0(nameServerAddr, question, additional, promise);
     }
 
     private Future<AddressedEnvelope<DnsResponse, InetSocketAddress>> query0(
             InetSocketAddress nameServerAddr, DnsQuestion question,
+            List<DnsRecord> additional,
```

Comment:
Yeah, it would be great if we could use clean for-loop here, but I guess it doesn't triumph the clean API design. Let's just use `Iterable` and optimize this further if the iteration ever happens more than once.